src/settings/CredentialsSettings.js

```diff
@@ -29,14 +31,30 @@ const CredentialsSettings = () => {
     updateCredentials(values);
   };
 
+  if (isLoading) {
+    return (
+      <Layout className="marginTop1">
+        <Icon icon="spinner-ellipsis" width="10px" />
+      </Layout>
+    );
+  }
+
+  if (error) {
+    return (
+      <div style={{ padding: '1rem' }}>
+        <p>
```

Comment:
Consider adding `role="alert" aria-live="polite"` around the error message, to improve accessibility.

Reply:
Updated to use `Loading` and `MessageBanner` from `stripes-components` for consistent styling and accessability.